src/scripts/options/settings.ts

```diff
@@ -38,6 +38,9 @@ export const toData = (useDefaults = false) => {
     } else if (els[key].type === 'checkbox') {
       val = !useDefaults ? (els[key] as HTMLInputElement).checked : !!val;
     }
+    if (els[key].classList.contains('enhancer-text-input') && val) {
+      els[key].classList.add('has-value');
+    }
```

Comment:
You need to remove the class again:
```suggestion
    if (els[key].classList.contains('enhancer-text-input')) {
      els[key].classList.toggle('has-value', !!val);
    }
```